### PR TITLE
Memory optimization: support responses bigger than max allowed memory

### PR DESCRIPTION
**Pull request by petrkotek**
## What?
Use temporary file for multi-status responses with 10000+ entries (& flush every 10000 entries).

## Why?
`$w->outputMemory()` doesn't work well for large responses (around 10MB, but may depend on php settings). ⚠️ Truncated response may be returned. ⚠️ 

Note 1: existing code works perfectly fine for responses <10MB)
Note 2: code inspired by http://codeinthehole.com/writing/creating-large-xml-files-with-php/

### CODE REVIEW

File: lib/DAV/Server.php [hunk 1]
```diff
@@ -1651,7 +1651,7 @@ function getResourceTypeForNode(INode $node) {
      *
      * @param array|\Traversable $fileProperties The list with nodes
      * @param bool $strip404s
-     * @return string
+     * @return string|resource String for less than 10000 entries. For more entries, returns a resource (file handler).
```
Comment (petrkotek) [inline lib/DAV/Server.php]: it's safe to return `resource` as result of `generateMultiStatus()` is always used as an argument to `$response->setBody()`, which accepts both `string` & `resource`.

**Comment by evert** [comment]
Instead of this approach, I would actually really like to refactor the sabre/http library a bit to allow callbacks to be specified as setBody(). Outputting the body would then call the function that was passed, which is then responsible for echoing to STDOUT.

This avoids the need for a temporary file, and makes this truly streaming instead of buffered. So for that reason I'd reject this, as I think this is 2 step forwards and 1 step back ;)

**Comment by petrkotek** [comment]
Sounds like a bit more work, however cleaner & more efficient indeed. Let me have a look in that.

**Comment by petrkotek** [comment]
groundwork in `sabre/http` to support callbacks as a body: https://github.com/fruux/sabre-http/pull/65

**Comment by evert** [comment]
Closing this as the sabre/http approach is under way.

**Pull request closed without merging**